control-plane-operator/hostedclusterconfigoperator/controllers/resources/kas/admissionpolicies.go

```diff
@@ -0,0 +1,150 @@
+package kas
+
+import (
+	"context"
+	"fmt"
+
+	hyperv1 "github.com/openshift/hypershift/api/hypershift/v1beta1"
+	"github.com/openshift/hypershift/control-plane-operator/hostedclusterconfigoperator/controllers/resources/manifests"
+	"github.com/openshift/hypershift/support/config"
+	"github.com/openshift/hypershift/support/upsert"
+	k8sadmissionv1beta1 "k8s.io/api/admissionregistration/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/ptr"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type AdmissionPolicy struct {
+	Name             string
+	MatchConstraints *k8sadmissionv1beta1.MatchResources
+	Validations      []k8sadmissionv1beta1.Validation
+}
+
+const (
+	AdmissionPolicyNameConfig = "config"
+	AdmissionPolicyNameICSP   = "icsp"
+)
+
+var (
+	defaultAdmissionPoliciesOperations = []k8sadmissionv1beta1.OperationType{"UPDATE", "DELETE"}
+	defaultMatchResourcesScope         = k8sadmissionv1beta1.ScopeType("*")
+	defaultMatchPolicyType             = k8sadmissionv1beta1.Equivalent
+	HCCOUserValidation                 = k8sadmissionv1beta1.Validation{
+		Expression: fmt.Sprintf("request.userInfo.username.contains('system:%s') || object.spec == oldObject.spec", config.HCCOUser),
+		Message:    "This resource cannot be created, updated, or deleted. Please ask your administrator to modify the resource in the HostedCluster object.",
+		Reason:     ptr.To(metav1.StatusReasonInvalid),
+	}
+)
+
+// ReconcileKASValidatingAdmissionPolicies will create ValidatingAdmissionPolicies which block certain resources
+// from being updated/deleted from the DataPlane side.
+func ReconcileKASValidatingAdmissionPolicies(ctx context.Context, hcp *hyperv1.HostedControlPlane, client client.Client, createOrUpdate upsert.CreateOrUpdateFN) error {
+	log := ctrl.LoggerFrom(ctx)
+	log.Info("reconciling validating admission policies")
+
+	if err := reconcileConfigValidatingAdmissionPolicy(ctx, hcp, client, createOrUpdate); err != nil {
+		return fmt.Errorf("failed to reconcile Config Validating Admission Policy: %v", err)
+	}
+
+	if err := reconcileICSPValidatingAdmissionPolicy(ctx, client, createOrUpdate); err != nil {
+		return fmt.Errorf("failed to reconcile ICSP Validating Admission Policy: %v", err)
+	}
+
+	return nil
+}
+
+func reconcileICSPValidatingAdmissionPolicy(ctx context.Context, client client.Client, createOrUpdate upsert.CreateOrUpdateFN) error {
+	icspAdmissionPolicy := AdmissionPolicy{Name: AdmissionPolicyNameICSP}
+	icspApiVersion := []string{"v1alpha1"}
```

Comment:
nit: s/Api/API/

You can also use the constants from the api itself:
```
import ( operatorv1alpha1 "github.com/openshift/api/operator/v1alpha1" )
...
icspAPIVersion := []string{operatorv1alpha1.GroupVersion.Version}
icspAPIGroup := []string{operatorv1alpha1.GroupVersion.Group}
```

Reply:
makes sense